data/transactions/logic/simulate_test.go

```diff
@@ -0,0 +1,635 @@
+					Receiver: sender2,
+					Amount:   basics.MicroAlgos{Raw: 1000000},
+				},
+			},
+		},
+		{
+			Txn: transactions.Transaction{
+				Type:   protocol.PaymentTx,
+				Header: makeBasicTxnHeader(sender2),
+				PaymentTxnFields: transactions.PaymentTxnFields{
+					Receiver: sender1,
+					Amount:   basics.MicroAlgos{Raw: 0},
+				},
+			},
+		},
+	}
+
+	// Should fail if there is no group parameter
+	result, err := s.SimulateSignedTxGroup(txgroup)
+	require.NoError(t, err)
+	require.Contains(t, *result.FailureMessage, "had zero Group but was submitted in a group of 2")
+
+	// Add group parameter
+	attachGroupID(txgroup)
+
+	// Check balances before transaction
+	sender1Data, _, err := l.LookupWithoutRewards(l.Latest(), sender1)
+	require.NoError(t, err)
+	require.Equal(t, basics.MicroAlgos{Raw: 1000000000}, sender1Data.MicroAlgos)
+
+	sender2Data, _, err := l.LookupWithoutRewards(l.Latest(), sender2)
+	require.NoError(t, err)
+	require.Equal(t, basics.MicroAlgos{Raw: 0}, sender2Data.MicroAlgos)
+
+	// Should now pass
+	result, err = s.SimulateSignedTxGroup(txgroup)
+	require.NoError(t, err)
+	require.Empty(t, result.FailureMessage)
+
+	// Confirm balances have not changed
+	sender1Data, _, err = l.LookupWithoutRewards(l.Latest(), sender1)
+	require.NoError(t, err)
+	require.Equal(t, basics.MicroAlgos{Raw: 1000000000}, sender1Data.MicroAlgos)
+
+	sender2Data, _, err = l.LookupWithoutRewards(l.Latest(), sender2)
+	require.NoError(t, err)
+	require.Equal(t, basics.MicroAlgos{Raw: 0}, sender2Data.MicroAlgos)
+}
+
+const trivialAVMProgram = `#pragma version 2
+int 1`
+
+func TestSimpleAppCall(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	l := makeSimulationTestLedger()
+	s := v2.MakeSimulator(l)
+
+	accounts := makeTestAccounts()
+	sender := accounts[0].Address
+
+	// Compile AVM program
+	ops, err := AssembleString(trivialAVMProgram)
+	require.NoError(t, err, ops.Errors)
+	prog := ops.Program
+
+	// Create program and call it
+	futureAppID := 1
+	txgroup := []transactions.SignedTxn{
+		{
+			Txn: transactions.Transaction{
+				Type:   protocol.ApplicationCallTx,
+				Header: makeBasicTxnHeader(sender),
+				ApplicationCallTxnFields: transactions.ApplicationCallTxnFields{
+					ApplicationID:     0,
+					ApprovalProgram:   prog,
+					ClearStateProgram: prog,
+					LocalStateSchema: basics.StateSchema{
+						NumUint:      0,
+						NumByteSlice: 0,
+					},
+					GlobalStateSchema: basics.StateSchema{
+						NumUint:      0,
+						NumByteSlice: 0,
+					},
+				},
+			},
+		},
+		{
+			Txn: transactions.Transaction{
+				Type:   protocol.ApplicationCallTx,
+				Header: makeBasicTxnHeader(sender),
+				ApplicationCallTxnFields: transactions.ApplicationCallTxnFields{
+					ApplicationID:     basics.AppIndex(futureAppID),
+					ApprovalProgram:   prog,
+					ClearStateProgram: prog,
+				},
+			},
+		},
+	}
+
+	attachGroupID(txgroup)
+	result, err := s.SimulateSignedTxGroup(txgroup)
+	require.NoError(t, err)
```

Comment:
I'd like to see another case here that verifies a rejecting app call has the right simulation result.